### PR TITLE
Ports Advanced Cameras

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports TG's Advanced Cameras

#### What are Advanced Cameras?

They're a type of camera console only usable by human mobs that effectively allows you to look around the station like an AI could--that is to say "free look" style. They lay an essential groundwork piece for Abductors.

These are not buildable by default and none are on station.

#### Also ports TG's Xenobiology Advanced Cameras

This type of advanced camera console is the same as the above advanced camera, only the camera range is limited to Xenobiology lab and any area that has the named "Xenobiology Lab". Not only does it allow you to look around, but it also allows you to pick up and drop slimes---you can also pick up dead monkies or feed monkey cubes into the console and drop monkies in.

This is currently not buildable or on station, but could very easily be mapped in later (would rather not cause conflicts with the two major refactors at the moment).

**Again, neither of the above  consoles are accessible or made buildable in this PR.**

#### Related

Adds in new cerulean slime reaction; it generates a single use set of blueprints that colors all the turfs in that area an lavenderish blue color--primarily intended to be used with the Xenobio camera system to rename areas to "Xenobiology Lab".

:cl: Fox McCloud
add: Implements Advanced Camera Consoles. A type of camera console that function like AI's vision style. Non-buildable/accessible.
add: Adds in Xenobiology Advance Cameras. A type of camera console that functions like advanced cameras, but only works in Xenobiology areas; can move around slimes and feed them with the console as well. Also non-buildable/accessible (for now).
add: Adds in cerulean slime reaction that generates a single use blueprint. On use it colors turfs a lavender color
/:cl:

### CODE REVIEW

File: code/game/area/Space Station 13 areas.dm [hunk 1]
```diff
@@ -1788,8 +1788,8 @@ area/security/podbay
 	icon_state = "toxlab"
 
 /area/toxins/xenobiology
-	name = "\improper Xenobiology Lab"
-	icon_state = "toxlab"
+	name = "Xenobiology Lab"
```
Comment (tigercat2000) [inline code/game/area/Space Station 13 areas.dm]: should have kept the \improper

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -438,6 +438,18 @@
 	max_amount = 60
 	turf_type = /turf/simulated/floor/sepia
 
+/obj/item/areaeditor/blueprints/slime
+	name = "cerulean prints"
+	desc = "A one use set of blueprints made of jelly like organic material. Renaming an area to 'Xenobiology Lab' will extend the reach of the management console."
+	color = "#2956B2"
+
+/obj/item/areaeditor/blueprints/slime/edit_area()
+	var/success = ..()
```
Comment (Tastyfish) [inline code/modules/research/xenobiology/xenobiology.dm]: Shouldn't this be `. = ..()`?

File: code/modules/research/xenobiology/xenobio_camera.dm [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+/datum/action/slime_place/Activate()
+	if(!target || !ishuman(owner))
+		return
+	var/mob/living/carbon/human/C = owner
+	var/mob/camera/aiEye/remote/xenobio/remote_eye = C.remote_control
+	var/obj/machinery/computer/camera_advanced/xenobio/X = target
+
+	if(cameranet.checkTurfVis(remote_eye.loc))
+		for(var/mob/living/carbon/slime/S in X.stored_slimes)
+			S.loc = remote_eye.loc
```
Comment (Tastyfish) [inline code/modules/research/xenobiology/xenobio_camera.dm]: forceMove

File: code/modules/research/xenobiology/xenobio_camera.dm [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+	if(cameranet.checkTurfVis(remote_eye.loc))
+		for(var/mob/living/carbon/slime/S in remote_eye.loc)
+			if(X.stored_slimes.len >= X.max_slimes)
+				break
+			if(!S.ckey)
+				if(S.buckled)
+					S.buckled.unbuckle_mob()
+				S.Feedstop()
+				S.visible_message("[S] vanishes in a flash of light!")
+				S.loc = X
```
Comment (Tastyfish) [inline code/modules/research/xenobiology/xenobio_camera.dm]: forceMove

File: code/modules/reagents/oldchem/chemical_reaction/chemical_reaction_slime.dm [hunk 5]
```diff
@@ -492,6 +492,20 @@
...
+		id = "s_territory"
+		result = null
+		required_reagents = list("blood" = 1)
+		result_amount = 1
+		required_container = /obj/item/slime_extract/cerulean
+		required_other = 1
+		on_reaction(datum/reagents/holder)
+			feedback_add_details("slime_cores_used","[type]")
+			var/obj/item/areaeditor/blueprints/slime/P = new /obj/item/areaeditor/blueprints/slime
+			P.loc = get_turf(holder.my_atom)
```
Comment (Tastyfish) [inline code/modules/reagents/oldchem/chemical_reaction/chemical_reaction_slime.dm]: forceMove

**Pull request merged**